### DISCUSSION TITLE
Reduce the number of steps to add an uninstalled behavior to an object

### DISCUSSION THREAD

**Discussion by tristanbob** [4ian/GDevelop#3632] [Enhancement ideas (and their technical discussions)]
## Description

When a user wants to add an uninstalled behavior to an object, they must perform a lot of steps.

![Screenshot 2021-09-25 172157](https://user-images.githubusercontent.com/8879811/134788405-fc2cfcaf-6407-4413-a97f-21767eb213d4.png)

![Screenshot 2021-09-25 172237](https://user-images.githubusercontent.com/8879811/134788410-26075f9a-16be-417f-9534-22eeeada4cb0.png)

![Screenshot 2021-09-25 172248](https://user-images.githubusercontent.com/8879811/134788413-c15d22b8-9df7-43a6-b940-ee8b13819cb3.png)

This process includes several unnecessary steps and it can easily confuse new users.  Here is the entire process:

1. Double-click object (or right-click "Edit object")
2. Click on the "Behaviors" tab
3. Click "Add a behavior to the object"
4. Enter a search string and hit enter
5. When the behavior is not found, click on the "Search New Behaviors" tab
6. Enter the same search string a second time
7. Select desired behavior
8. Click "Install in project"
9. Click "Back" (or hit "Escape")
10. Click "Installed behaviors" tab
11. Select the newly installed extension
12. Select "Add  a behavior to the object"

## Solution suggested

Simplify the process of adding a behavior to an object.  Here is the proposed workflow:

1. Double-click object (or right-click "Edit object")
2. Click on the "Behaviors" tab
3. Click "Add a behavior to the object"
4. Enter a search string and hit enter
5. Select the desired behavior (All extensions will be listed regardless if they are installed or not.  There could be a filter option to only display "Installed" extensions)
6. Select "Add  a behavior to the object" (The extension will be installed, if needed, and added to the object.  The user will be back at the "Behaviors" tab of the object they are editing.

## Alternatives considered

One quick improvement to the current experience would be to copy the search string used in "Installed behaviors" so that it automatically filters extensions in the "Search New Behaviors" tab.

**Comment by 4ian** [comment]
In short: yep, definitely need to improve this process because it's needlessly cumbersome.
